portal-impl/src/com/liferay/portal/servlet/I18nServlet.java

```diff
@@ -178,6 +182,8 @@ protected I18nData getI18nData(HttpServletRequest httpServletRequest)
 					friendlyURL);
 
 				if (siteGroup == null) {
+					httpServletRequest.setAttribute("I18nErrorPath", i18nPath);
```

Comment:
Can you create a constant for "I18nErrorPath"? The name should be something like I18N_ERROR_PATH (and the value "I18N_ERROR_PATH"; same capitalization).